### ISSUE TITLE
First Species Clefs

### ISSUE THREAD

**Issue by joelrust** [artusiinc/issues#13]
In First Species, the clefs do not adjust to the _cantus firmus'_ range:

![image](https://user-images.githubusercontent.com/35672157/36206514-a89b2bee-1160-11e8-9e28-a9ac002a4753.png)

This can be exaggerated by the transposing mechanism which doesn't always send it to the closest transposition:

![image](https://user-images.githubusercontent.com/35672157/36206613-07e37494-1161-11e8-90fb-174486df7d42.png)

**Comment by ryaanahmed** [comment]
The clef of the CF is constrained by the option "Cantus firmus staff" in the section editor. The options are either treble or bass, and the student counterpoint goes on whichever you don't choose.

There are a bunch of things that this means you can't do, some of which we'll likely implement:
- the system doesn't intelligently pick a clef based on the range (as you've indicated here)
- you can't have questions with the CFs on both the top and bottom line within the same section

Current workaround is to make a separate section for CFs that you want on the treble clef.

We'll work on the transposition thing and make that better re: which direction to transpose!

**Comment by mscuthbert** [comment]
Yes, we can change the c.f. staff from treble/bass to top/bottom and then allow instructors to choose the clef for each staff, so that if you like to have students do species counterpoint in alto clef (or soprano / tenor) you can.  Getting intelligent clef choices will need to be a longer-range project since it adds a lot of instructor complexity to set up what clefs they have taught their students already, etc.
